### PR TITLE
Add KinD deployment and remote debugging support

### PR DESCRIPTION
**Pull request by ssh2go**

### CODE REVIEW

File: atlas/commands/bootstrap/bootstrap.go [hunk 1]
```diff
@@ -49,6 +51,8 @@ var (
 	initializeProfiler = initialize.Bool(flagWithProfiler, false, "initialize the application with a profiling service")
 	initializeHelm     = initialize.Bool(flagWithHelm, false, "initialize the application with the helm charts")
 	initializeExpand   = initialize.String(flagExpandName, "", "the name of the input file for the `expand` command (optional)")
+	initializeKind     = initialize.Bool(flagWithKind, false, "initialize the application with KinD support (optional, only applicable when helm charts are enabled)")
```
Comment (drewwells) [inline atlas/commands/bootstrap/bootstrap.go:54]: The atlas-cli seems to have a lot of helpful things turned off like `helm` and `kind`. Perhaps we should make these default==true?

File: atlas/templates/kind/.id.gotmpl [hunk 2]
```diff
@@ -0,0 +1 @@
+{{ .Helm.GetName }}
```
Comment (drewwells) [inline atlas/templates/kind/.id.gotmpl:1]: This file shouldn't be included. It's intended as a per developer identifier for determining namespace to install to.

Comment (drewwells) [reply atlas/templates/kind/.id.gotmpl:1]: We also need to update https://github.com/infobloxopen/atlas-cli/blob/master/atlas/templates/.gitignore.gotmpl to ignore file named `/.id`

File: atlas/application/application.go [hunk 3]
```diff
@@ -165,10 +182,34 @@ func (app Application) generateFile(filename, templatePath string) error {
 	return err
 }
 
+func (app Application) generateId() error {
+	return app.generateFile("kind/.id", "templates/kind/.id.gotmpl")
+}
+
```
Comment (drewwells) [inline atlas/application/application.go:188]:
```suggestion
```

File: atlas/templates/Makefile.kind.gotmpl [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+IMAGE_VERSION    := $(shell git describe --long --tags --dirty=-unreleased --always || echo pre-commit)
+REGISTRY         := {{ .Registry }}
+IMAGE_REPO       := {{ .Name }}
+DOCKER_IMAGE     := $(REGISTRY)/$(IMAGE_REPO):$(IMAGE_VERSION)
+NODE_IMAGE       ?= kindest/node:v${NODE_VERSION}
+KUBECONFIG       ?= ${HOME}/.kube/config
+KUBECTL          := bin/kubectl
+KIND             := bin/kind
+HELM             := bin/helm
+TARGET_NAMESPACE ?= `cat kind/.id`
```
Comment (drewwells) [inline atlas/templates/Makefile.kind.gotmpl:16]: Put this file in the root, it's used generally for `helm` usecases and isn't specific to kind

```suggestion
TARGET_NAMESPACE ?= `cat .id`
```

File: atlas/templates/Makefile.kind.gotmpl [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+HELM3_VER        ?= `helm version -c --template {{`{{.Version}}`}} | grep v3`
+KERNEL_NAME      := `uname`
+
+.id:
+	@git config user.email | awk -F@ '{print $$1}' > kind/.id
```
Comment (drewwells) [inline atlas/templates/Makefile.kind.gotmpl:21]:
```suggestion
	@git config user.email | awk -F@ '{print $$1}' > .id
```

**Review by drewwells — DISMISSED** [review]
I think .id makes more sense at the root. Some other recommendations otherwise looks good

File: atlas/commands/bootstrap/bootstrap.go (hunk 1 above)
Comment (ssh2go) [reply atlas/commands/bootstrap/bootstrap.go:54]: We have a lot of disabled features here, and it's not only the last-added things... You're right, we have a lot of extra functionality here, but probably it makes sense these features were enabled in accordance with user requirements, by this user, during the service creation. I assume some devs don't need helmcharts, kind or something else, especially if they are going to use their services as a simple/ standalone apps or just a sandboxes.
BTW, just an idea: in theory, we can create config file and store it somewhere in the system, but I'm not sure regarding the usability. For example, that file will be filled first time when the user creates his first service using atlas-cli, command-line args will be used as these "preferred values". After that, we can use that config as default if no args provided, but command-line args can override these values.

**Comment by ssh2go** [comment]
> I think .id makes more sense at the root. Some other recommendations otherwise looks good

I added all the files related to KinD into the according folder, that's why the `.id` file is also there. `.id` file has not been used in Helm before, however the cli-tool supported it. So, for now this file is used only for KinD purposes. Of course we can move it to root, but I'm not sure if somebody will use it separately too often...

File: atlas/templates/kind/.id.gotmpl (hunk 2 above)
Comment (ssh2go) [reply atlas/templates/kind/.id.gotmpl:1]: Yeah, user can remove this file and generate new one based on his git config, however I assume the service name as a "default" namespace also is suitable. Considering that, should we remove this file anyway?

File: atlas/commands/bootstrap/bootstrap.go (hunk 1 above)
Comment (drewwells) [reply atlas/commands/bootstrap/bootstrap.go:54]: Config file is a good idea for future work 🎫

**Review by drewwells — APPROVED** [review]

**Pull request merged**